### PR TITLE
add await on i18n translations to generate function for wirters

### PR DESCRIPTION
**Pull request by kjose90**
#2048

Added `await initI18n()` to generate function of `fiori-elements-writer` and `fiori-freestyle-writer` to load i18n translations asynchronously to ensure proper initialisation.

- added new configuration to `lrop.test.ts` in `packages/fiori-elements-writer/test/lrop.test.ts`
- added new configuration to `basic.test.ts` in `packages/fiori-freestyle-writer/test/basic.test.ts`

Added new configuration to ensure i18n translations are correctly loaded and mocked during app generation.
Existing tests lacked verification of i18n translations when `localOnly` and `generateIndex` are true in `getPackageJsonTasks` functionality, hence these checks were added.

### CODE REVIEW

File: packages/fiori-freestyle-writer/test/basic.test.ts [hunk 1]
```diff
@@ -152,6 +152,18 @@ describe(`Fiori freestyle template: ${TEST_NAME}`, () => {
                 appOptions: { generateIndex: false }
             },
             settings: {}
+        },
+        {
+            name: 'basic_witht_start-noflp',
```
Comment (IainSAP) [inline packages/fiori-freestyle-writer/test/basic.test.ts:157]: typo?

**Review by IainSAP — APPROVED** [review]
Thanks @kjose90 

- Updated as expected
- We could also remove the loading of i18n here: https://github.com/SAP/open-ux-tools/blob/main/packages/fiori-elements-writer/src/i18n.ts#L35 as its no longer needed but should not do any harm

**Review by IainSAP — APPROVED** [review]
Re-approving

**Comment by kjose90** [comment]
Thanks everyone.

**Pull request merged**